### PR TITLE
deps: Bump next and related dependencies

### PR DESCRIPTION
**Pull request by lforst**
When bumping next, Vercel's NFT logic seems to have changed and we were pulling in weird additional files like `.git/objects` and webpack caches. We identified that this was due to `mdx.ts` using `process.cwd()` and path concatenations which was confusing NFT, making it think pretty much everything in the root of the repo was needed to run the app. This caused us to be over the 300 MB limit for AWS lambdas.

Entails:
- All kinds of related patch and minor bumps to Next.js and MDX
- Minor QoL improvement in vercel.json
- A more statically analyzable `src/mdx.ts` file

Fixes https://github.com/getsentry/sentry-docs/issues/10491

### CODE REVIEW

**Comment by lforst** [comment]
From 180MB

Screenshot 2024-06-26 at 12 52 29

to 15MB

Screenshot 2024-06-26 at 12 52 38

File: package.json [hunk 1]
```diff
@@ -24,7 +24,8 @@
     "lint:prettier:fix": "prettier --write \"./{src,app,scripts}/**/*.{md,mdx,ts,tsx,js,jsx,mjs}\"",
     "lint:fix": "yarn run lint:prettier:fix && yarn run lint:eslint:fix",
     "sidecar": "yarn spotlight-sidecar",
-    "test": "jest"
+    "test": "jest",
+    "build-trace": "npx next@canary internal turbo-trace-server .next/trace"
```
Comment (chargome) [inline package.json:28]: Should we keep that?

**Review by chargome — APPROVED** [review]
🚀

**Comment by chargome** [comment]
Fixes #10491

File: src/mdx.ts [hunk 2]
```diff
@@ -96,8 +98,8 @@ async function getDocsFrontMatterUncached(): Promise<FrontMatter[]> {
   return frontMatter;
 }
 
-export function getAllFilesFrontMatter(folder: string = 'docs') {
```
Comment (a-hariti) [inline src/mdx.ts:99]: this `folder` shouldn't be hardcoded as we need to pass it as a parameter for developer docs for example later on

Comment (lforst) [reply src/mdx.ts:99]: Right now it doesn't need to be. We can add it back if needed. We shouldn't add code for the mere possibility of needing it.

Comment (a-hariti) [reply src/mdx.ts:99]: I just mentioned it since this is deleting something that was already there, but I see what you mean

**Review by oioki — APPROVED** [review]

**Pull request merged**